gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/JSPHookProjectProvider.java

```diff
@@ -0,0 +1,217 @@
+
+        IPath location = PathBridge.create( op.getLocation().content() );
+
+        final String hostBundle = op.getCustomOSGiBundle().content();
+
+        IPath temp = GradleCore.getDefault().getStateLocation().append( hostBundle );
+
+        try
+        {
+            ZipUtil.unzip( new File( op.getRealOSGiBundleFile().content() ), temp.toFile() );
+        }
+        catch( IOException e1 )
+        {
+            e1.printStackTrace();
+        }
+
+        String bundleSymbolicName = "";
+        String version = "";
+
+        if( temp.toFile().exists() )
+        {
+            File file = temp.append( "META-INF" ).append( "MANIFEST.MF" ).toFile();
+            String[] contents = FileUtil.readLinesFromFile( file );
+
+            for( String content : contents )
+            {
+                if( content.contains( "Bundle-SymbolicName:" ) )
+                {
+                    bundleSymbolicName =
+                        content.substring( content.indexOf( "Bundle-SymbolicName:" ) + "Bundle-SymbolicName:".length() );
+                }
+
+                if( content.contains( "Bundle-Version:" ) )
+                {
+                    version = content.substring( content.indexOf( "Bundle-Version:" ) + "Bundle-Version:".length() );
+                }
+            }
+        }
+
+        ElementList<OSGiCustomJSP> jsps = op.getCustomJSPs();
+
+        StringBuilder sb = new StringBuilder();
+        sb.append( "create " );
+        sb.append( "-d \"" + location.toFile().getAbsolutePath() + "\" " );
+        sb.append( "-t " + "jsphook" + " " );
+
+        if( !bundleSymbolicName.equals( "" ) )
+        {
+            sb.append( "-h " + bundleSymbolicName + " " );
+        }
+
+        if( !version.equals( "" ) )
+        {
+            sb.append( "-H " + version + " " );
+        }
+
+        sb.append( "\"" + projectName + "\" " );
+
+        try
+        {
+            final String[] ret = BladeCLI.execute( sb.toString() );
+
+            final String errors = BladeCLI.checkForErrors( ret );
+
+            if( errors.length() > 0 )
+            {
+                retval = GradleCore.createErrorStatus( "Project create error: " + errors );
+                return retval;
+            }
+
+            IPath projecLocation = location;
+
+            final String lastSegment = location.lastSegment();
+
+            if( location != null && location.segmentCount() > 0 )
+            {
+                if( !lastSegment.equals( projectName ) )
+                {
+                    projecLocation = location.append( projectName );
+                }
+            }
+
+            for( OSGiCustomJSP jsp : jsps )
+            {
+                File jspFile = temp.append( jsp.getValue().content() ).toFile();
+
+                if( jspFile.exists() )
+                {
+                    String parent = jspFile.getParentFile().getPath();;
+                    String metaInfResources = "META-INF\\resources";
```

Comment:
This is the code that is failling for Linux and Mac.  \\ doesn't work as a path.. just use "META-INF/resources" since that should work for both.